src/device/r4300/interrupt.c

```diff
@@ -460,6 +449,9 @@ void nmi_int_handler(void* opaque)
     g_gs_vi_counter = 0;
     init_interrupt(&r4300->cp0);
 
+    dev->vi.next_vi = cp0_regs[CP0_COUNT_REG] + dev->vi.delay;
```

Comment:
Sorry I'm not at my computer for the next few days, does setting next_vi to 0 work here (and don't schedule any interrupt). I would think that just setting next_vi to 0 should still allow the VI interrupts to restart, the method you have here might cause them to restart too soon

Reply:
Just setting next_vi = 0 (and removing the subsequent add_interrupt_event_cont) still breaks the reset unfortunately.